tools/docker/image-pull.sh

```diff
@@ -59,12 +59,12 @@ main() {
     dbg "TAG=$TAG"
     dbg "rootdir=$rootdir"
 
-    # The tag used in dockerhub is the base image with : converted to -
-    upstream_tag="$(echo "$IMAGE" | tr : -)"
-    info "Base docker image $IMAGE -> tag $tag"
+    # The registry used in Gitlab has the base image name appended with : converted to -
```

Comment:
NACK this. At least for me, I didn't intend to drop dockerhub and switch to the gitlab registry. Instead, what I meant was to:
- dockerbuild images as part of CI;
- push them to the registry in each pipeline;
- push them to dockerhub on master.

So the image-pull.sh script wouldn't be affected. Instead, CI should not use image-pull.sh but pull from the registry "manually". Or, better yet, not pull anything explicitly and rely on automatic pulling, by passing the image name explicitly to run.sh.

But that's just my opinion, maybe dropping dockerhub is simpler and good enough.

Reply:
> Instead, what I meant was to:

    dockerbuild images as part of CI;
    push them to the registry in each pipeline;
    push them to dockerhub on master.

This is almost what I had in mind, only that last step would have been: tag the image with "latest" on Gitlab if we're on the master branch instead of pushing to dockerhub.
It doesn't really make a difference to me though, the only thing we need to push to dockerhub on master is a robot account/token if we don't already have one, but it shouldn't be a big deal.

One change that I propose to keep though, is to tag the images like `prplmesh-runner-ubuntu-18.04` instead of `prplmesh-runner:ubuntu-18
04`. 
On Gitlab since we're going to have hundreds of different tags (one for each pipeline) I think it makes things easier if the different variants are not mixed together under a single registry name.

Reply:
As discussed orally: we agree.